BlueprintUI/Sources/Layout/LayoutAttributes.swift

```diff
@@ -153,7 +153,34 @@ public struct LayoutAttributes {
     private func validateAlpha() {
         assert(alpha.isFinite, "LayoutAttributes.alpha must only contain finite values.")
     }
-    
+
+    /// Performs rounding on the frame to snap to pixel boundaries.
+    ///
+    /// - Parameters:
+    ///   - origin: The global origin to offset the frame by before rounding. This offset is used to ensure that
+    ///     positive and negative frame coordinates both round away from zero.
+    ///   - correction: The amount of rounding correction to apply to the origin before rounding, to account for the
+    ///     rounding applied to this node's parent.
+    ///   - scale: The screen scale to use when rounding.
+    mutating func round(from origin: CGPoint, correction: CGPoint, scale: CGFloat) -> CGPoint {
```

Comment:
Is it worth making `RoundingCorrection` its own type that wraps a `CGPoint`? Seeing all these CGPoints and CGFloats makes it a bit hard to follow what's going on, IMO.

Reply:
I could see that if it were publicly exposed, but as an internal thing I think the documentation is probably sufficient.